eureka/S4_generate_lightcurves/s4_genLC.py

```diff
@@ -144,91 +149,122 @@ def lcJWST(eventlabel, ecf_path='./', s3_meta=None):
                 meta.nspecchan = len(meta.wave_hi)
-            meta.wave_hi = np.array(meta.wave_hi)
             meta.wave_low = np.array(meta.wave_low)
+            meta.wave_hi = np.array(meta.wave_hi)
+            meta.wave = (meta.wave_low + meta.wave_hi)/2
+            meta.wave_err = (meta.wave_hi - meta.wave_low)/2
 
             if not hasattr(meta, 'boundary'):
                 meta.boundary = 'extend' # The default value before this was added as an option
 
+            # Create masked array for steps below
+            optspec_ma = np.ma.masked_array(spec.optspec, spec.optmask)
+            # Create opterr array with same mask as optspec
+            # FINDME: This implementation is terrible, masked arrays aren't working here
+            opterr_ma = np.ma.copy(optspec_ma)
+            opterr_ma = spec.opterr
+
             # Do 1D sigma clipping (along time axis) on unbinned spectra
-            optspec = np.ma.masked_array(optspec)
             if meta.sigma_clip:
-                log.writelog('Sigma clipping unbinned spectral time series', mute=(not meta.verbose))
+                log.writelog('Sigma clipping unbinned optimal spectra along time axis')
                 outliers = 0
                 for l in range(meta.subnx):
-                    optspec[:,l], nout = clipping.clip_outliers(optspec[:,l], log, wave_1d[l], meta.sigma, meta.box_width, meta.maxiters, meta.boundary, meta.fill_value, verbose=meta.verbose)
+                    optspec_ma[:,l], nout = clipping.clip_outliers(optspec_ma[:,l], log, spec.wave_1d[l], meta.sigma, meta.box_width, meta.maxiters, meta.boundary, meta.fill_value, verbose=meta.verbose)
                     outliers += nout
-                log.writelog('Identified a total of {} outliers in time series, or an average of {} outliers per wavelength'.format(outliers, np.round(outliers/meta.subnx, 1)), mute=meta.verbose)
+                # Print summary if not verbose
+                log.writelog('Identified a total of {} outliers in time series, or an average of {:.3f} outliers per wavelength'.format(outliers, outliers/meta.subnx), mute=meta.verbose)
 
             # Apply 1D drift/jitter correction
             if meta.correctDrift:
                 #Calculate drift over all frames and non-destructive reads
-                log.writelog('Applying drift/jitter correction', mute=(not meta.verbose))
+                log.writelog('Applying drift/jitter correction')
                 # Compute drift/jitter
-                meta = drift.spec1D(optspec, meta, log)
+                drift1d, driftmask = drift.spec1D(optspec_ma, meta, log)
+                # Replace masked points with moving mean
+                drift1d = clipping.replace_moving_mean(drift1d, driftmask, Box1DKernel(meta.box_width))
+                spec['drift1d'] = (['time'], drift1d)
+                spec['driftmask'] = (['time'], driftmask)
```

Comment:
I'd suggest we add this to lc instead of spec since one might want to use it as a covariate.